group-coordinator/src/main/java/org/apache/kafka/coordinator/group/classic/ClassicGroup.java

```diff
@@ -1300,6 +1341,68 @@ public Map<String, byte[]> groupAssignment() {
         ));
     }
 
+    /**
+     * Convert the current classic group to a consumer group.
+     * Add the records for the conversion.
+     *
+     * @param consumerGroup     The converted consumer group.
+     * @param records           The list to which the new records are added.
+     *
+     * @throws GroupIdNotFoundException if any of the group's member doesn't support the consumer protocol.
+     */
+    public void convertToConsumerGroup(
+        ConsumerGroup consumerGroup,
+        List<Record> records,
+        TopicsImage topicsImage
+    ) throws GroupIdNotFoundException {
+        consumerGroup.setGroupEpoch(generationId);
+        consumerGroup.setTargetAssignmentEpoch(generationId);
```

Comment:
nit: From a structure perspective, it would be great if we could pair the records creation with the group update. It will be easier to read.